rclrs/src/node/graph.rs

```diff
@@ -0,0 +1,568 @@
+        };
+        Ok(count)
+    }
+
+    /// Count the number of subscribers for a given topic
+    pub fn count_subscribers(&self, topic: &str) -> Result<usize, RclrsError> {
+        let topic_name = CString::new(topic).unwrap();
+        let mut count: usize = 0;
+
+        // SAFETY: No preconditions for this function
+        unsafe {
+            rcl_count_subscribers(&*self.rcl_node_mtx.lock(), topic_name.as_ptr(), &mut count)
+                .ok()?
+        };
+        Ok(count)
+    }
+
+    /// Get topic publisher info
+    pub fn get_publishers_info_by_topic(
+        &self,
+        topic: &str,
+    ) -> Result<Vec<TopicEndpointInfo>, RclrsError> {
+        self.get_publisher_subscriber_info_by_topic(topic, rcl_get_publishers_info_by_topic)
+    }
+
+    /// Get topic subscriber info
+    pub fn get_subscribers_info_by_topic(
+        &self,
+        topic: &str,
+    ) -> Result<Vec<TopicEndpointInfo>, RclrsError> {
+        self.get_publisher_subscriber_info_by_topic(topic, rcl_get_subscriptions_info_by_topic)
+    }
+
+    /// Get an rcl names_and_types function, with a "demangle" argument
+    fn get_names_and_types_by_node_with_demangle_arg(
+        &self,
+        node: &str,
+        namespace: &str,
+        getter: unsafe extern "C" fn(
+            *const rcl_node_t,
+            *mut rcl_allocator_t,
+            bool,
+            *const ::std::os::raw::c_char,
+            *const ::std::os::raw::c_char,
+            *mut rcl_names_and_types_t,
+        ) -> rcl_ret_t,
+    ) -> Result<TopicNamesAndTypes, RclrsError> {
+        let mut rcl_names_and_types = unsafe { rmw_get_zero_initialized_names_and_types() };
+        let node_name = CString::new(node).unwrap();
+        let node_namespace = CString::new(namespace).unwrap();
+        unsafe {
+            getter(
+                &*self.rcl_node_mtx.lock() as *const _,
+                &mut rcutils_get_default_allocator(),
+                false,
+                node_name.as_ptr(),
+                node_namespace.as_ptr(),
+                &mut rcl_names_and_types,
+            )
+        };
+
+        convert_names_and_types(rcl_names_and_types)
+    }
+
+    /// Get an rcl names_and_types function, without a "demangle" argument
+    fn get_names_and_types_by_node_without_demangle_arg(
+        &self,
+        node: &str,
+        namespace: &str,
+        getter: unsafe extern "C" fn(
+            *const rcl_node_t,
+            *mut rcl_allocator_t,
+            *const ::std::os::raw::c_char,
+            *const ::std::os::raw::c_char,
+            *mut rcl_names_and_types_t,
+        ) -> rcl_ret_t,
+    ) -> Result<TopicNamesAndTypes, RclrsError> {
+        let mut rcl_names_and_types = unsafe { rmw_get_zero_initialized_names_and_types() };
+        let node_name = CString::new(node).unwrap();
+        let node_namespace = CString::new(namespace).unwrap();
+        unsafe {
+            getter(
+                &*self.rcl_node_mtx.lock() as *const _,
```

Comment:
I think you might need to rebase onto main, this now should require `.lock().unwrap()` since we switched to std lib mutexes.